### ISSUE TITLE
An extension for VSCode for syntax highlighting

### ISSUE THREAD

**Issue by BlockOG** [pfusik/cito#64]

**Comment by Danon** [comment]
And maybe for IntelliJ as well

**Comment by pfusik** [comment]
I started developing the VSCode extension.

**Comment by pfusik** [comment]
![image](https://user-images.githubusercontent.com/351822/176154177-9d5302f6-fbee-48f2-8692-711a138498ca.png)

Still working on it. You can try it by opening VSCode as follows:

    code --extensionDevelopmentPath=/full/path/to/editors/vscode

**Comment by pfusik** [comment]
I added [syntax highlighting](https://github.com/pfusik/cito/blob/master/editors/scite/ci.properties) for [SciTE](https://www.scintilla.org/SciTE.html), my favorite editor for Windows/Mac/Linux.

**Comment by pfusik** [comment]
Released to the Marketplace:
![image](https://user-images.githubusercontent.com/351822/201108532-63a10caf-86d7-4919-b3b3-747e46146a94.png)
Syntax highlighting only for now.

Also released `cito` as a .NET tool. Install with:

    dotnet tool install -g cito

and follow the displayed instructions to add to your PATH.

**Comment by Danon** [comment]
@pfusik Planning on plugin for JetBrains IDE?

**Comment by pfusik** [comment]
@Danon Yes. This requires a parser in Java, which is mostly done (#48).

**Comment by Danon** [comment]
@pfusik Do you have it open source?

**Comment by pfusik** [comment]
The VSCode extension now reports syntax errors, using the parser in Ć transpiled to JavaScript:

![vscode-ci-syntax-errors](https://user-images.githubusercontent.com/351822/215065074-1ef129c4-1f4f-4d44-a00c-3561db1793d5.png)

To be released soon.

**Comment by pfusik** [comment]
I've just released Fusion 3.2.0, which includes major changes in the VSCode extension. Now it reports all errors, not only syntax errors. "Go to definition" works. More features (autocomplete etc) to come.

![image](https://github.com/fusionlanguage/fut/assets/351822/27e3a369-aa7a-4bac-b37d-0848f5abd58e)

**Comment by SergioFLS** [comment]
Are there any plans to publish the extension to [Open VSX Registry](https://open-vsx.org/)? It would make the extension easily available on VSCode-derived editors (like VSCodium).

**Comment by pfusik** [comment]
I wasn't aware of Open VSX Registry. [Published now](https://open-vsx.org/extension/fusionlanguage/fusion) - please test!

I also added syntax highlighting for Visual Studio (non-Code) and Vim: https://github.com/fusionlanguage/fut/blob/master/doc/editors.md

**Comment by jemoo** [comment]
This is so cool. I really need the auto-completion function. Is it in progress?

**Comment by pfusik** [comment]
Eric Lippert estimated C# auto-completion to [10 man-years](https://stackoverflow.com/a/9556530/2032514). :)

**Comment by jemoo** [comment]
I think perhaps just providing auto-completion of type names, field names, and method names is enough to greatly improve development efficiency.

**Comment by jemoo** [comment]
> Eric Lippert estimated C# auto-completion at [10 man-years](https://stackoverflow.com/a/9556530/2032514). :)

I think perhaps just providing auto-completion of type names, field names, and method names is enough to greatly improve development efficiency.

**Comment by pfusik** [comment]
> I think perhaps just providing auto-completion of type names, field names, and method names is enough to greatly improve development efficiency.

What other auto-completions are there?

**Comment by jemoo** [comment]
> > I think perhaps just providing auto-completion of type names, field names, and method names is enough to greatly improve development efficiency.
> 
> What other auto-completions are there?

I think there may also be local variables, function signatures (parameters), context-aware keywords, statement completion etc. (perhaps my description is not professional enough).

**Comment by SergioFLS** [comment]
> I wasn't aware of Open VSX Registry. [Published now](https://open-vsx.org/extension/fusionlanguage/fusion) - please test!

Tried it out, seems to work just fine. Thank you very much!

**Comment by pfusik** [comment]
I'm open to feedback on the VSCode extension as I'm not using VSCode personally. What works well, what needs to be added (aside from auto-completion, which is high-priority) ? Any bugs?

**Comment by pfusik** [comment]
Closing.
The VSCode has been in the [Marketplace](https://marketplace.visualstudio.com/items?itemName=fusionlanguage.fusion) for three years now. It's also released to [Open VSX Registry](https://open-vsx.org/extension/fusionlanguage/fusion) and works with Cursor and Windsurf.
A major missing feature is auto-complete, which I want to work on in the coming months.

I don't plan to work on IntelliJ.
